src/caches/adams_bashforth_moulton_caches.jl

```diff
@@ -322,17 +324,22 @@ du_cache(c::VCAB3Cache) = ()
 function alg_cache(alg::VCAB3,u,rate_prototype,uEltypeNoUnits,uBottomEltypeNoUnits,tTypeNoUnits,uprev,uprev2,f,t,dt,reltol,p,calck,::Type{Val{false}})
   k2 = rate_prototype
   k3 = rate_prototype
-  ϕstar_nm1 = zeros(typeof(rate_prototype),3)
   dts = zeros(typeof(dt),3)
   c = zeros(typeof(t), 3, 3)
   g = zeros(typeof(t), 3)
-  ϕ_n = zeros(typeof(rate_prototype),3)
-  ϕstar_n = zeros(typeof(rate_prototype),3)
+  ϕ_n = Vector{typeof(rate_prototype)}(3)
+  ϕstar_nm1 = Vector{typeof(rate_prototype)}(3)
+  ϕstar_n = Vector{typeof(rate_prototype)}(3)
+  for i in 1:3
+    ϕ_n[i] = rate_prototype
```

Comment:
This is going to alias the same array to each of these spots. Does that not cause an issue further down?

Reply:
https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/e50f73d42ea8f5c52e6651170daa4cccc21d0699/src/adams_utils.jl#L17-L24

This will have problems if ϕ_n is an array of all the same array since that loop will essentially not mean anything. How are tests passing with this? And why not use `ϕ_n[i] = zeros(rate_prototype)`?

Reply:
for `out of place`,type of `rate_prototype` would not be `arrayType`

Reply:
but all `perform_step!` for `out-place` have written without considering it as `arrayType`, I think

Reply:
no, and for most methods there's a test for it.

Reply:
could you elaborate, any example method I can look?

Reply:
I mean, if you take a look at any of them:

https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/perform_step/fixed_timestep_perform_step.jl#L235-L265

you'll notice that these will work with `u` an array. It won't be the fastest way because of memory allocations, but it will work.

Reply:
oh, got it but then we have also take care of pointers to arrays(in case of passing to other function etc)

Reply:
if `k1,k2,...` are `arrayType`, they should pass by value not ref in other function, how should I  take care of it?

Reply:
in which function call will that be an issue?

Reply:
checking `arrayType` for every `vars` and then passes it by ref or value in `perform_step!`??

Reply:
it shouldn't matter. Point to the code where it would matter.

Reply:
https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/perform_step/adams_bashforth_moulton_perform_step.jl#L481

Reply:
I see, yes the AB codes will need a few changes to make this work well. But add the `copy` here as a first change in that direction, and make note of this in an issue. That means just edit the capitalization in test description and this is good to merge.